akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala

```diff
@@ -109,21 +110,80 @@ class Http2ServerDemux extends GraphStage[BidiShape[Http2SubStream, FrameEvent,
       }
 
       // we should not handle streams later than the GOAWAY told us about with lastStreamId
+      private var continuation: Boolean = false
+      private var closing: Int = 0
       private var closedAfter: Option[Int] = None
       private var incomingStreams = mutable.Map.empty[Int, SubStream]
       private var totalOutboundWindowLeft = Http2Protocol.InitialWindowSize
       private var streamLevelWindow = Http2Protocol.InitialWindowSize
+      // TODO: make me configurable
+      private val streamTimeout = 100.milliseconds
+      private val connectionTimeout = streamTimeout.plus(streamTimeout)
 
       def lastStreamId: Int = {
         incomingStreams.lastOption.map(_._1).getOrElse(0)
       }
 
-      def pushGOAWAY(errorCode: ErrorCode = Http2Protocol.ErrorCode.PROTOCOL_ERROR): Unit = {
-        // http://httpwg.org/specs/rfc7540.html#rfc.section.6.8
-        val last = lastStreamId
-        closedAfter = Some(last)
-        bufferedFrameOut.push(GoAwayFrame(last, errorCode))
-        // FIXME: handle the connection closing according to the specification
+      def connectionClose(): Unit = {
```

Comment:
well this looks non working in tests, so maybe somebody could enlighten me.
I actually tested it with https://github.com/summerwind/h2spec but actually I'm not sure if this also tests the termination of the tcp connection?!